File: 2023/day3/part1.py
lines = []
with open("input.txt") as f:
	lines = f.read().split('\n')[:-1]

#lines = ['467..114..', '...*......', '..35..633.', '......#...', '617*......', '.....+.58.', '..592.....', '......755.', '...$.....*', '.664...598']

sm = 0

for j in range(len(lines)):
	d = ""
	f = False
	line = lines[j]
	for i in range(len(line)):
		if line[i].isdigit():
			d += line[i]
			if i > 0:
				if not line[i-1].isdigit() and line[i-1] != '.':
					f = True
				elif j > 0 and not lines[j-1][i-1].isdigit() and lines[j-1][i-1] != '.':
					f = True
				elif j < len(lines) - 1 and not lines[j+1][i-1].isdigit() and lines[j+1][i-1] != '.':
					f = True
			if i < len(line) - 1:
				if not line[i+1].isdigit() and line[i+1] != '.':
					f = True
				elif j > 0 and not lines[j-1][i+1].isdigit() and lines[j-1][i+1] != '.':
					f = True
				elif j < len(lines) - 1 and not lines[j+1][i+1].isdigit() and lines[j+1][i+1] != '.':
					f = True
			if j > 0 and not lines[j-1][i].isdigit() and lines[j-1][i] != '.':
				f = True
			if j < len(lines) - 1 and not lines[j+1][i].isdigit() and lines[j+1][i] != '.':
				f = True
		elif len(d) and f:
			sm += int(d)
			d = ""
			f = False
		else:
			d = ""

	if len(d) and f:
		sm += int(d)

print(sm)

File: 2024/day1/part1.cpp
#include <vector>
#include <string>
#include <ctype.h>
#include <cstdlib>
#include <fstream>
#include <iostream>
#include <algorithm>

int main()
{
	int	i;
	int	j;
	int	len;

	std::string line;
	std::string last;
	std::string first;
	std::vector<int> lst1;
	std::vector<int> lst2;

	std::ifstream input("input.txt");

	if (!input.is_open())
	{
		std::cout << "cannot open input.txt" << std::endl;
		return (1);
	}

	len = 0;
	while (getline(input, line))
	{
		i = 0;
		while (line[i] && isdigit(line[i]))
			i++;
		
		first = line.substr(0, i);

		while (line[i] && !isdigit(line[i]))
			i++;

		last = line.substr(i);

		lst1.push_back(atoi(first.c_str()));
		lst2.push_back(atoi(last.c_str()));

		len++;
	}

	std::sort(lst1.begin(), lst1.end());
	std::sort(lst2.begin(), lst2.end());

	i = 0;
	j = 0;
	while (i < len)
	{
		j += abs(lst1[i] - lst2[i]);
		i++;
	}

	std::cout << "total distance = " << j << std::endl;

	input.close();
}
